src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs

```diff
@@ -189,9 +190,17 @@ private void InitializePackagesFolderInfo(string packagesFolderPath, bool exclud
             IEnumerable<SourceRepository> secondarySources,
             CancellationToken token)
         {
+            var signedPackageVerifier = new PackageSignatureVerifier(SignatureVerificationProviderFactory.GetSignatureVerificationProviders());
+
             var downloadContext = new PackageDownloadContext(resolutionContext.SourceCacheContext)
             {
-                ParentId = nuGetProjectContext.OperationId
+                ParentId = nuGetProjectContext.OperationId,
+                ExtractionContext = new PackageExtractionContext(
+                    PackageSaveMode.Defaultv3,
```

Comment:
Please make sure `downloadContext` is always being used to extract packages in global packages folder that's where `DefaultV3` is applicable. If this is ever used for solution packages folder, then it will change the existing behavior.